### PR TITLE
Add autofix support for stdin input

### PR DESCRIPTION
**Pull request by mahmost**
> Which issue, if any, is this issue related to?

#2710 

> Is there anything in the PR that needs further explanation?

No, it's self explanatory.

### CODE REVIEW

**Comment by evilebottnawi** [comment]
@mahmost Thanks!

**Comment by mahmost** [comment]
Since maintainers seem busy at the moment .. I'm willing to add some more information to help reviewing the PR

Here is what was done :
- In case `--fix` is used with stdin input, the output (to stdout) is the fixed style code
- One simple test was added to autofix related tests ensuring that when autofixing stdin code, `result.output` contains the fixed style code

Other additions and consideration that was **NOT** done in this PR :
- Outputting the normal output (linting rules) to stderr (this may even be an option `--also-output-rules` or anything like that)
- More tests (I just looked at the other autofix tests and added what I thought to be enough for now)

File: lib/standalone.js [hunk 1]
```diff
@@ -118,7 +118,15 @@ module.exports = function(
       })
       .catch(handleError)
       .then(stylelintResult => {
-        return prepareReturnValue([stylelintResult]);
+        const returnValue = prepareReturnValue([stylelintResult]);
+        const postcssResult = stylelintResult._postcssResult;
+        // If we're fixing the output should be the fixed code
```
Comment (ntwb) [inline lib/standalone.js]: `// If we're fixing, the output should be the fixed code` the added comma here helps readability

**Comment by ntwb** [comment]
Thanks @mahmost, see the above for one change to the code comment and I'd like to see a few more tests added if you could please?

I'm thinking in particular a test or two thats tests the output formatting when `{`, `}`, `\r`, `\n` are used with some of the _new line before/after opening/closing braces_ rules

**Comment by dmitriy-lodyanov** [comment]
Good day
Guys, any progress here?
Maybe i can help with final stage of this pr?

**Comment by evilebottnawi** [comment]
@mahmost friendly ping

**Comment by modosc** [comment]
i'm working on this now.

**Comment by modosc** [comment]
> I'm thinking in particular a test or two thats tests the output formatting when `{`, `}`, `\r`, `\n` are used with some of the _new line before/after opening/closing braces_ rules

@ntwb afaict there aren't `--fix` rules for any of those - they're still on the list in #2829. if we add the comma and rebase is that enough?

**Comment by dmitriy-lodyanov** [comment]
@mahmost Thanks!

**Comment by modosc** [comment]
@stylelint/core ping - sorry for the wide net, can anyone respond to the above comments? would love to get this going forward.

**Comment by CAYdenberg** [comment]
As the autofixed rules are minimal for now and none of them affect things like line breaks, I recommend we merge with the single test and address any issues that crop up moving forward. Any thoughts @ntwb ?

@modosc With stale PRs (and simple changes) I usually make them myself and push to the contributors branch (if possible) or open a new PR. In this case would you like to do the honours?

**Comment by mahmost** [comment]
Sorry for not giving this enough time, the latest commit addresses the requested comment change, and handles ignored files (by not returning anything if file is ignored)
Still, the issues of new lines and spaces seem to persist for now

**Comment by modosc** [comment]
> As the autofixed rules are minimal for now and none of them affect things like line breaks, I recommend we merge with the single test and address any issues that crop up moving forward. Any thoughts @ntwb ?

let's go ahead with this and once we start fixing newlines we can add appropriate tests (i understand the concern around the output in that case for sure).

**Pull request merged**